src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+#if DEBUG
+using System.Runtime.CompilerServices;
+#endif
+
+using SafeCurlHandle = Interop.libcurl.SafeCurlHandle;
+using curlioerr = Interop.libcurl.curlioerr;
+using curliocmd = Interop.libcurl.curliocmd;
+
+namespace System.Net.Http
+{
+    internal partial class CurlHandler
+    {
+        private static readonly char[] s_LineEndings = new char[] { HttpRuleParser.CR, HttpRuleParser.LF };
+        private static readonly string s_LineEndingDelimiter = "\r\n";
+        private ulong CurlReceiveHeadersCallback(
```

Comment:
I hadn't made these static thinking it would be easy to call CheckDisposed(). But I see now from the other comment that this approach is unsafe